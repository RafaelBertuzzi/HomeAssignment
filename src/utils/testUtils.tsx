import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
