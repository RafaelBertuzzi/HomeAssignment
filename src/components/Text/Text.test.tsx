import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { Text } from '.';

describe('Component: Text', () => {
  afterEach(cleanup);

  it('should render text', () => {
    const json = rendererWithProviders(<Text title="test title" />);
    expect(json).toMatchSnapshot();
  });
});
