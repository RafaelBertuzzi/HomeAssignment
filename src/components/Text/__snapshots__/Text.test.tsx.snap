// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Text should renders text 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    dataSet={{}}
    style={
      {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
  >
    test title
  </Text>
</RNCSafeAreaProvider>
`;
